ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -13,137 +12,145 @@ import { filterOptions, defaultMatcher } from 'ember-power-select/utils/group-ut
+  @tracked showModal = false;
+  @tracked createdRecord = null;
+  @tracked modelData = null;
 
-  get inputValue() {
-    return this.args.inputValue || [];
+  get hidePowerSelect() {
+    return this.selectedOptions.length >= this.args.selectLimit;
   }
 
   get shouldRenderName() {
-    return this.args.shouldRenderName || false;
+    return this.args.models?.some((model) => model.includes('identity')) || this.args.shouldRenderName
+      ? true
+      : false;
   }
 
-  get excludeOptions() {
-    return this.args.excludeOptions || null;
+  addSearchText(optionsToFormat) {
+    // maps over array models from query
+    return optionsToFormat.toArray().map((option) => {
+      option.searchText = `${option.name} ${option.id}`;
+      return option;
+    });
   }
 
-  get passObject() {
-    return this.args.passObject || false;
+  formatInputAndUpdateDropdown(inputValues) {
+    // inputValues are initially an array of strings from @inputValue
+    // map over so selectedOptions are objects
+    return inputValues.map((option) => {
+      let matchingOption = this.dropdownOptions.findBy('id', option);
+      // remove any matches from dropdown list
+      this.dropdownOptions.removeObject(matchingOption);
+      return {
+        id: option,
+        name: matchingOption ? matchingOption.name : option,
+        searchText: matchingOption ? matchingOption.searchText : option,
+      };
+    });
   }
 
-  @action
-  async fetchOptions() {
-    try {
-      let queryOptions = {};
-      let options = await this.store.query(this.args.model, queryOptions);
-      this.formatOptions(options);
-    } catch (err) {
-      if (err.httpStatus === 404) {
-        if (!this.allOptions) {
-          // If the call failed but the resource has items
-          // from a different namespace, this allows the
-          // selected items to display
-          this.allOptions = [];
+  @task
+  *fetchOptions() {
+    this.dropdownOptions = []; // reset dropdown anytime we re-fetch
+    if (!this.args.models) {
+      return;
+    }
+
+    for (let modelType of this.args.models) {
+      try {
+        let queryParams = {};
+        // fetch options from the store
+        let options = yield this.store.query(modelType, queryParams);
+        if (this.args.excludeOptions) {
+          options = options.filter((o) => !this.args.excludeOptions.includes(o.id));
         }
-        return;
-      }
-      if (err.httpStatus === 403) {
-        this.shouldUseFallback = true;
-        return;
+        // add to dropdown options
+        this.dropdownOptions = [...this.dropdownOptions, ...this.addSearchText(options)];
+      } catch (err) {
+        if (err.httpStatus === 404) {
+          // continue to query other models even if one 404s
+          // and so selectedOptions will be set after for loop
+          continue;
+        }
+        if (err.httpStatus === 403) {
+          if (this.dropdownOptions && this.args.models.length > 1) continue;
```

Comment:
`RGP` policies are enterprise only, so this renders the fallback component even if a user has access to list/read `ACL` policies. 

I don't love this way of handling it and open to alternative ways to handle it. 

This works since `acl` is the first model passed in so if that call is successful and populates `dropdownOptions`, then the component will render a search select of just ACL policies

Reply:
I think since we are planning a refactor, if it works for your use case I would leave it as is.

Reply:
I think this makes sense, to continue getting other models before rendering the fallback. Maybe we leave a comment here to that end?